ui/pages/projects/[projectid]/builds/[buildid].js

```diff
@@ -0,0 +1,68 @@
+import React, { useCallback, useEffect, useState } from 'react';
```

Comment:
some funny filenames here. What sort of black magic is this? :p

Reply:
this is thankfully fully handled by nextjs - when a page on the corresponding level is rendered, the `projectid` and `buildid` will be passed as parameters to the render function, so for this file and an address like `/projects/123/builds/465` the function will get `{ projectid: '123', buildid: '465' }`...